src/algebra/big_operators.lean

```diff
@@ -432,6 +432,15 @@ lemma prod_nat_pow (s : finset α) (n : ℕ) (f : α → ℕ) :
 by haveI := classical.dec_eq α; exact
 finset.induction_on s (by simp) (by simp [nat.mul_pow] {contextual := tt})
 
+lemma prod_flip {n : ℕ} (f : ℕ → β) :
+  (range (nat.succ n)).prod (λ r, f (n - r)) = (range (nat.succ n)).prod f :=
+begin
+  induction n with n ih,
```

Comment:
You should have one instruction per line, and different goals should be enclosed by braces (here and in the rest of the file).